File: 02-Backend/src/main/java/com/mayaank/ecommerce/entity/OrderItem.java
package com.mayaank.ecommerce.entity;

import jakarta.persistence.*;

import java.math.BigDecimal;

@Entity
@Table(name = "order_item")
public class OrderItem {
    @Id
    @GeneratedValue(strategy = jakarta.persistence.GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;
@Column(name = "image_url")
    private String imageUrl;
@Column(name = "quantity")
    private int quantity;
@Column(name = "unit_price")
    private BigDecimal unitPrice;
@Column(name = "product_id")
    private Long  productId;
@Column(name = "order_id")
    private Order order;


}

File: 03-Frontend/my-app/src/app/common/order-item.ts
import { CardItem } from "./card-item";

export class OrderItem {
    imageUrl: string;
    unitPrice: number;
    quantity: number;
    productId: number;

    constructor(cartItem: CardItem) {
        this.imageUrl = cartItem.imageUrl;
        this.unitPrice = cartItem.unitPrice;
        this.quantity = cartItem.quantity;
        this.productId = cartItem.id;
    }
}
